src/softmax.jl

```diff
@@ -19,6 +16,12 @@ independent.
 """
 softmax(xs) = softmax!(similar(xs), xs)
 
+function softmax(xs::AbstractArray{T}; dims=1) where {T}
+    max = maximum(xs, dims=dims)
+    out = exp.(xs .- max)
+    out = out ./ sum(out, dims=dims)
```

Comment:
I think it'd be easier to write `out` in one line here (with the benefit of fusing everything).

Also, if you add this method we'll need to remove the first `softmax(xs)` method, since they overlap in signature.

Otherwise looking good!